### PR TITLE
Fallback to stub protoadapters when java.time is missing

### PR DESCRIPTION
**Pull request by oldergod**
This allows clients, specially Android ones, not to be forced to
to address the issue if they don't have to. If a client is
actually using google.protobuf.Duration or g.p.Instant, the
generated code will not compile and clients will be able to
detect the missing parts at compile time, instead of having
the stub proto adapter failing at runtime.

### CODE REVIEW

File: wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/ProtoAdapter.kt [hunk 1]
```diff
@@ -251,12 +251,37 @@ actual abstract class ProtoAdapter<E> actual constructor(
...
+
+    /**
+     * Stub [ProtoAdapter] for Wire types which are typeliased to `java.time` types on the JVM
+     * such as [Duration] and [Instant]. This proto adapter is used when the corresponding
+     * `java.time` type is missing from the JVM classpath.
+     */
+    class UnsupportedTypeProtoAdapter : ProtoAdapter<Nothing>(FieldEncoding.LENGTH_DELIMITED,
+        Nothing::class) {
+      override fun redact(value: Nothing) =
+          throw IllegalStateException("Operation not supported.")
```
Comment (JakeWharton) [inline wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/ProtoAdapter.kt:278]: Feels like we should mention core library desugaring or minSdk 26 here.

**Comment by oldergod** [comment]
I'll try setting up some tests.

**Comment by JakeWharton** [comment]
For Retrofit I just copied a smoke JVM test into an Android module and added API 21, 24, 26, and 29 to the GitHub Action. You should be able to mostly do that here as well.

**Comment by JakeWharton** [comment]
I also don't think it's a blocker. Test on Cash and if it works merge so it makes the beta release. We can land tests over time.

**Comment by oldergod** [comment]
Yes, on Cash!

**Pull request merged**